### ISSUE TITLE
Swarm mode - support sticky sessions

### ISSUE THREAD

**Issue by foleymic** [containous/traefik#1024]
### Problem Statement ###
Currently, under swarm mode, sticky sessions won't work because Traefik only registers the docker service instead of the containers.  This means that if you have a service which has **_x_** replicas Traefik will only register the service and not know about the underlying **_x_** containers (or tasks).  Instead it will rely on Docker Swarm's built in load balancer.  For most modern applications, this is ok, but for applications that, unfortunately, require session persistence, or sticky sessions, this won't work.  Traefik will only maintain stickiness to the service, which will then use Docker Swarm's built in Load Balancer to route the request to a task.  

For this to work, Traefik needs to know about the underlying containers, or tasks, which make up the service and direct requests to specific containers on the overlay network versus letting Swarm direct requests to arbitrary containers that are part of this service.  Because Swarm's load balancer operates at the transport layer (layer 4) instead of layer 7, this essentially means that there is no way for Docker to support cookie based session persistence  (and IP based persistence is not a viable solution because a lot of companies go through a proxy, which means one IP being seen by the load balancer could be thousands of users).

### Proposed Enhancement ###
This enhancement is to add a feature which, under swarm mode, will allow Traefik to essentially bypass Docker's load balancer and instead let Traefik do the load balancing and direct the requests to the tasks under the service on the overlay network versus the service itself.

pseudo code:
docker.go
```
listServices{

....

   for _, service := range serviceList {
      if swarm mode && traefik.backend.loadbalancer.sticky==true or traefik.backend.loadbalancer.diableSwarmLB==true {
        dockerDataListTasks, err = listTasks(ctx, dockerClient, service.ID, dockerData)
        ....
     }
   }
}
```

**Comment by Tester98** [comment]
I would upvote sticky sessions, however the current method of injecting cookies is the not the best way. the load balancer, should feature hashing based on say ipv4 or ipv6 hash as an option for sticky session

**Comment by foleymic** [comment]
@Tester98 - I don't disagree with what you are saying, but that would have to be part of a seperate ticket.  I'm not changing anything here in how Traefik handles sticky sessions, I'm just allowing for their support when operating under Swarm mode.

**Comment by ldez** [comment]
Closed by #1033
